src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -325,6 +325,11 @@ internal void RegisterBuiltInSerializers()
             Register(typeof(IList<>));
             Register(typeof(IDictionary<,>));
             Register(typeof(IEnumerable<>));
+            Register(typeof(ICollection<>));
+            Register(typeof(ISet<>));
+            Register(typeof(IReadOnlyDictionary<,>));
+            Register(typeof(IReadOnlyCollection<>));
+            Register(typeof(IReadOnlyList<>));
```

Comment:
This is breaking if used for persistance or versioning. Are we OK including it in the patch release?